File: src/main/java/profitAnalysis.java
import java.util.Arrays;
import java.util.List;

public class profitAnalysis {

    int countlongestPeriodProfit(List<Integer> input){
        int max = 0;
        int currentMax =0;
        for(Integer element:input){
             if(element==1){
                 currentMax++;
             }else{
                 if(currentMax>max){
                     System.out.println(" current max "+ currentMax + " max "+ max);
                     max = currentMax;
                     currentMax = 0;
                 }
             }
        }
         if(currentMax>max)return currentMax; else return max;
    }

    List<String> longestMonths(List<Integer> input,List<String> months){
        int start=0,end =input.size();
        int counter =0 ,max =0;
        for(int index=0; index<input.size(); index++){
            if(input.get(index) ==1){
                counter++;
            }else {
                if(counter > max ){
                    max =counter;
                    counter = 0 ;
                    end = index;
                }
            }

        }
        if(counter > max ) max =counter;
        System.out.println(" end "+ end + " max " +max);
        return Arrays.asList(months.get(end-max),months.get(end));
    }

    public static void main(String[] args) {
        profitAnalysis testSubject = new profitAnalysis();
        System.out.println(testSubject.countlongestPeriodProfit(Arrays.asList(1, 0, 1, 1, 0, 1, 1, 1, 0, 1))== 3);
        //(1,0,1,1,1,1,1,1,0))
//        System.out.println(testSubject.countlongestPeriodProfit(Arrays.asList(1,0,1,1,1,1,1,1,0))== 6);
//        System.out.println(testSubject.countlongestPeriodProfit(Arrays.asList(0,0))== 0);
//        System.out.println(testSubject.countlongestPeriodProfit(Arrays.asList(1,1))== 2);

        List<String> months = Arrays.asList("Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct");
        System.out.println(testSubject.longestMonths(Arrays.asList(1, 0, 1, 1, 0, 1, 1, 1, 0, 1),months));
        System.out.println(testSubject.longestMonths(Arrays.asList(1,0,1,1,1,1,1,1,0),months));
        System.out.println(testSubject.longestMonths(Arrays.asList(1,1,1,1,1,1,1),months));
    }
}

File: src/main/kotlin/ProfitAnalysis.kt
class ProfitAnalysis {

    fun countlongestPeriodProfit(profits: List<Int>) : Int{
        var maxProfitDuration = 0
        var counter = 0
        profits.forEach {
            if(it == 0){
                counter = 0
            }else{
                counter += 1
            }
            if(counter > maxProfitDuration){
                maxProfitDuration = counter
            }
        }
        return maxProfitDuration
    }

    fun findlongestProfitPeriod(profits: List<Int>, months: List<String>) : Output{
        var maxProfitDuration = 0
        var counter = 0

        var endIndex  = 0
        profits.forEachIndexed{index, it ->
            if(it == 0){
                counter = 0
            }else{
                counter += 1
            }
            if(counter > maxProfitDuration){
                maxProfitDuration = counter
                endIndex = index
            }
        }

        var result = Output(maxProfitDuration,"","")
        if(maxProfitDuration != 0){
            result = Output(maxProfitDuration, months[endIndex-maxProfitDuration+1],months[endIndex])
        }
        return result
    }
}

data class Output ( val count: Int, val startMonth: String, val endMonth: String)

fun main(args: Array<String>) {
    var testSubject = ProfitAnalysis()
    var months = listOf("Jan","Feb","Mar","Apr","May","Jun","Jul","Aug","Sep","Oct")
    println(testSubject.countlongestPeriodProfit(listOf(1, 0, 1, 1, 0, 1, 1, 1, 0, 1)) == 3)
    println(testSubject.countlongestPeriodProfit(listOf(1,0,1,1,1,1,1,1,0)) == 6)
    println(testSubject.countlongestPeriodProfit(listOf(0,0,0)) == 0)

    println(testSubject.countlongestPeriodProfit(listOf(1,1,1)) == 3)

    println(testSubject.countlongestPeriodProfit(emptyList()) == 0)
    //println(testSubject.countlongestPeriodProfit(listOf(1, 0, 1, 1, 0, 1, 1, 1, 0, 1)) == 0)

    println(testSubject.findlongestProfitPeriod(listOf(1,0,1,1,0,1,1,1,0,1),months).equals(Output(3,"Jun","Aug")))
    println(testSubject.findlongestProfitPeriod(listOf(0,0),months).equals(Output(0,"","")))
    println(testSubject.findlongestProfitPeriod(listOf(1),months).equals(Output(1,"Jan","Jan")))

}